File: src/python/5.Vererbung/Main.py
from Monster import Monster
from Vampir import Vampir
def main():
    golem = Monster("Golem", 1000,15)
    vampir = Vampir("Vampir", 150,10,10)
    zusammen =[]
    zusammen.append(golem)
    zusammen.append(vampir)
    while golem.hp>0 and vampir.hp>0:
        golem.attack(vampir)
        vampir.attack(golem)
        print(f"Beide haben noch Golem: {golem.hp} Vampir: {vampir.hp} leben")
    print("fertig")


if __name__=="__main__":
    main()

File: src/java/5.Vererbung/Main.java
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) {
        Monstera monster = new Monstera("Golem", 1000, 10);
        Vampir vampir = new Vampir("Vampir", 200, 10, 10);
        ArrayList<Monstera> monsters = new ArrayList<>();
        monsters.add(monster);
        monsters.add(vampir);

        while (monster.getHP() > 0 && vampir.getHP() > 0) {
            monster.attack(vampir);
            vampir.attack(monster);
            System.out.println("Es haben beide noch " + monster.getHP() + " " + vampir.getHP() + " Leben");
        }
        System.out.println("Fertig");
    }
}
